rero_ils/modules/items/api/circulation.py

```diff
@@ -434,13 +435,21 @@ def compare_item_pickup_transaction_libraries(self, **kwargs):
     @add_action_parameters_and_flush_indexes
     def checkout(self, current_loan, **kwargs):
         """Checkout item to the user."""
+        # Check if the 'checkout' action is allowed
```

Comment:
I think it is better to move this check to the beginning of the method `add_action_parameters_and_flush_indexes`.  you do not need that we execute the logic of `add_action_parameters_and_flush_indexes` and then come to the function to find out that we are not authorized to do a checkout

Reply:
Right! However, I think it's even better to create a new decorator `@check_operation` called before `@add_action_parameters_and_flush_indexes`.

Reply:
Yes, even better